### ISSUE TITLE
[Hold for payment][$500] Chat - Action menu for MP4 files has "Copy to clipboard" and "Edit comment" functions

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#35578]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.35-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause- Internal Team
**Slack conversation:**

## Action Performed:
Pre-requisite: user must be logged in.
1. Go to any chat.
2. Send a MP4 file.
3. Right click on it for the action menu to appear

## Expected Result:
"Copy to clipboard" and "Edit comment" should not be options for MP4 files.

## Actual Result:
"Copy to clipboard" and "Edit comment" are functions available for MP4 files. When edited, the file is no longer available.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidences>

https://github.com/Expensify/App/assets/93399543/637abd2e-8bf4-4d5c-ae58-e9e521548708

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01bd7a8a1a0e4bc08b</li>
        <li>Upwork Job ID: 1753109564515966976</li>
        <li>Last Price Increase: 2024-02-01</li>
<li>Automatic offers: </li>
<ul>
<li>DylanDylann | Reviewer | 0</li>
<li>Tony-MK | Contributor | 0</li>
</ul></li>
    </ul>
</details>

**Comment by kbecciv** [comment]
We think that this bug might be related to #vip-vsb
CC @quinthar

**Comment by Tony-MK** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Chat - Action menu for MP4 files has "Copy to clipboard" and "Edit comment" functions

### What is the root cause of that problem?

The root cause of the problem is that the `text` of a video will be the video's name but should be `[Attachment]`.

![Screenshot 2024-02-01 at 22 09 38](https://github.com/Expensify/App/assets/22982173/0957d133-0eba-4c9f-827c-400285724d54)

Therefore, the first condition seen below will fail for messages that are video attachments.

https://github.com/Expensify/App/blob/9ab4fe92dcbac8252998febf6fa24a080a196a70/src/libs/isReportMessageAttachment.ts#L20

Also, we did not test the [`isReportMessageAttachment`](https://github.com/Expensify/App/blob/9ab4fe92dcbac8252998febf6fa24a080a196a70/src/libs/isReportMessageAttachment.ts#L20) function with any other test but `[Attachment]` as text.

https://github.com/Expensify/App/blob/33463c2bdf0abe10de018e96ddbdfda01d582736/tests/unit/isReportMessageAttachmentTest.js#L3-L10

### What changes do you think we should make in order to solve the problem?


After matching the HTML with the `CONST.ATTACHMENT_UPLOADING_MESSAGE_HTML` regex, we should add another condition to check for any other attachment types that do not show a preview if the `[Attachment]` condition fails. E.g. `isVideo` for video attachments.

Finally, add [tests](https://github.com/Expensify/App/blob/9ab4fe92dcbac8252998febf6fa24a080a196a70/src/libs/isReportMessageAttachmentTest.ts) for the changes created.
### What alternative solutions did you explore? (Optional)
N/A

**Comment by Tony-MK** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Chat - Action menu for MP4 files has "Copy to clipboard" and "Edit comment" functions

### What is the root cause of that problem?

The root cause of the problem is that the `text` of a video will be the video's name but should be `[Attachment]`.

![Screenshot 2024-02-01 at 22 09 38](https://github.com/Expensify/App/assets/22982173/0957d133-0eba-4c9f-827c-400285724d54)

Therefore, the first condition seen below will fail for messages that are video attachments.

https://github.com/Expensify/App/blob/9ab4fe92dcbac8252998febf6fa24a080a196a70/src/libs/isReportMessageAttachment.ts#L19-L20

Also, we did not test the [`isReportMessageAttachment`](https://github.com/Expensify/App/blob/9ab4fe92dcbac8252998febf6fa24a080a196a70/src/libs/isReportMessageAttachment.ts#L20) function with any other test but `[Attachment]` as text.

https://github.com/Expensify/App/blob/33463c2bdf0abe10de018e96ddbdfda01d582736/tests/unit/isReportMessageAttachmentTest.js#L3-L10

### What changes do you think we should make in order to solve the problem?


After matching the HTML with the `CONST. ATTACHMENT_SOURCE_ATTRIBUTE ` regex, we should add another condition to check for any other attachment types that do not show a preview if the `[Attachment]` condition fails. E.g. `isVideo` for video attachments.

Finally, add [tests](https://github.com/Expensify/App/blob/9ab4fe92dcbac8252998febf6fa24a080a196a70/src/libs/isReportMessageAttachmentTest.ts) for the changes created.

### What alternative solutions did you explore? (Optional)
N/A

**Comment by DylanDylann** [comment]
I will review tomorrow

**Comment by DylanDylann** [comment]
Waiting for more proposal

**Comment by Tony-MK** [comment]
Hi @DylanDylann, I updated my [proposal](https://github.com/Expensify/App/issues/35578#issuecomment-1922085803).
Could you review it?
Thanks

**Comment by DylanDylann** [comment]
Thanks for everyone. @Krishna2323's [proposal](https://github.com/Expensify/App/issues/35578#issuecomment-1921879684) doesn't fix this issue completely (miss "Edit comment" option). @Tony-MK's [proposal](https://github.com/Expensify/App/issues/35578#issuecomment-1922085803)  looks good to me

🎀 👀 🎀 C+ reviewed

**Comment by DylanDylann** [comment]
> because canEditReportAction is also used in ReportScreen and [isReportMessageAttachment](https://github.com/Expensify/App/blob/9ab4fe92dcbac8252998febf6fa24a080a196a70/src/libs/isReportMessageAttachment.ts#L19-L20) is different from [isReportActionAttachment](https://github.com/Expensify/App/blob/afa88d45f670aebd0bde2a0045aed3dc0ef0e7e1/src/libs/ReportActionsUtils.ts#L670-L671).


I don't see any problem with modifying [isReportMessageAttachment](https://github.com/Expensify/App/blob/9ab4fe92dcbac8252998febf6fa24a080a196a70/src/libs/isReportMessageAttachment.ts#L19-L20) with [isReportActionAttachment](https://github.com/Expensify/App/blob/afa88d45f670aebd0bde2a0045aed3dc0ef0e7e1/src/libs/ReportActionsUtils.ts#L670-L671) in the canEditReportAction function

**Comment by srikarparsi** [comment]
Hey @DylanDylann, @Krishna2323 did seem to include the edit comment option in his proposal. Were there any other reasons you wanted to go with @Tony-MK's proposal?

**Comment by Tony-MK** [comment]
@Krishna2323, You edited your proposal numerously, and at the point of review, you did not specify the `editAction` function, which is crucial in this issue. Hence, I did not copy your proposal.

Furthermore, you introduced the `editAction` function just a day ago. A couple of minutes after @DylanDylann made his decision. That is not fair. A screenshot is below, but @srikarparsi, you should check the edit timeline of @Krishna2323's proposal to confirm this.

![Edit Timeline](https://github.com/Expensify/App/assets/22982173/0daad033-e1d9-4f67-bb3c-fab9ac25f88b)

![Screenshot 2024-02-09 at 08 41 44](https://github.com/Expensify/App/assets/22982173/e78adb09-69f0-4175-89a7-77d378e9fb5f)


Finally, not to be rude, but I backed off from this [issue](https://github.com/Expensify/App/issues/34835) which you were assigned where your RCA and solution were flawed. However, I respected the C+'s and the engineer's decision. Anyway, someone else did the right fix.

I don't want to debate because it is against the rules, but let's work as a team.

**Comment by Tony-MK** [comment]
@Krishna2323, You edited your proposal numerously, and at the point of review, you did not specify the `editAction` function, which is crucial in this issue. Hence, I did not copy your proposal.

Furthermore, you introduced the `editAction` function just a day ago. A couple of minutes after @DylanDylann made his decision. That is not fair. A screenshot is below, but @srikarparsi, you should check the edit timeline of @Krishna2323's proposal to confirm this.

![Edit Timeline](https://github.com/Expensify/App/assets/22982173/0daad033-e1d9-4f67-bb3c-fab9ac25f88b)

![Screenshot 2024-02-09 at 08 41 44](https://github.com/Expensify/App/assets/22982173/e78adb09-69f0-4175-89a7-77d378e9fb5f)


Finally, not to be rude, but I backed off from this [issue](https://github.com/Expensify/App/issues/34835) which you were assigned where your RCA was flawed. However, I respected the C+'s and the engineer's decision. Anyway, someone else did the right fix.

I don't want to debate because it is against the rules, but let's work as a team.

**Comment by Tony-MK** [comment]
> you edited your proposal 2 hours before the review

1. You mentioned it in the root cause but I haven't found your solution yet.
2. Your solution excludes `isAttachment`.
3. Furthermore, you did not even test your solution because it would not work.
4. How would you expect a C++ also to review your previous proposals? 🤔  (Your last proposal is your final and best version of your proposal). 
5. Over-edit your proposal, even after review.

These are the weaknesses of the proposal I noticed and only edited three times. 

I tested my solution and know it works. Yours doesn't. Simple.

**Comment by DylanDylann** [comment]
@srikarparsi When I reviewed the proposal, @Krishna2323 did not mention the edit comment option in his proposal. I approved @Tony-MK's proposal because that's the only one that fixes this issue completely.

@Krishna2323 Read the guideline carefully
https://github.com/Expensify/App/blob/9463e22ba5c0897d519c0ccc85c12ae591742541/contributingGuides/CONTRIBUTING.md?plain=1#L100-L108
 Please post a new comment when you edit your proposal to avoid confusion

**Comment by DylanDylann** [comment]
I gave my decision [here](https://github.com/Expensify/App/issues/35578#issuecomment-1933586349), the final decision is up to the internal engineer

**Comment by srikarparsi** [comment]
Ok, lets go with @Tony-MK in this case. I do understand that @Krishna2323 had edit in the root cause and this is very semantic but it could've been included in the changes. I also think we should

> Switch [isReportMessageAttachment](https://github.com/Expensify/App/blob/9ab4fe92dcbac8252998febf6fa24a080a196a70/src/libs/isReportMessageAttachment.ts#L19-L20) with [isReportActionAttachment](https://github.com/Expensify/App/blob/afa88d45f670aebd0bde2a0045aed3dc0ef0e7e1/src/libs/ReportActionsUtils.ts#L670-L671) in the canEditReportAction function.

as long as it doesn't cause any regressions.

**Comment by kadiealexander** [comment]
Not overdue, being worked on in the PR.

**Comment by DylanDylann** [comment]
@Tony-MK any update on the PR

**Comment by Tony-MK** [comment]
Sorry for the delay, @DylanDylann.

I am having issues testing android - mWeb. 

However, I should be done by tomorrow.

**Comment by techievivek** [comment]
@srikarparsi It seems like the backend is returning the filename instead of the `[Attachment]` text. Were you able to look into it? There is also a comment from @mkhutornyi regarding the same: https://github.com/Expensify/App/issues/35578#issuecomment-1921954417.

CC @francoisl since you were involved with this project. 🙇 


Also, it would be great to have this fixed as quickly as possible @DylanDylann @Tony-MK, since we had another issue created today that was marked as deploy blocker https://github.com/Expensify/App/issues/36828.

**Comment by DylanDylann** [comment]
@Tony-MK Please help to speed up our PR, because our PR will fix other deploy blocker issues

**Comment by Tony-MK** [comment]
@DylanDylann Sorry for the delay. Let's finish the PR.

**Comment by francoisl** [comment]
> It seems like the backend is returning the original filename instead of the [Attachment] text. Were you able to look into it? There is also a comment from

@techievivek I'm working on this already, we also need to make the backend return `[Attachment]` for the `lastMessageText` value when uploading a video.

**Comment by techievivek** [comment]
Ok great, thanks for confirming @francoisl 🙇

**Comment by DylanDylann** [comment]
@kadiealexander The PR was deployed to production 2 weeks ago. Please help to process the next step

**Comment by kadiealexander** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@DylanDylann] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@DylanDylann] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@DylanDylann] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@DylanDylann] Determine if we should create a regression test for this bug.
- [ ] [@DylanDylann] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@kadiealexander] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by kadiealexander** [comment]
Payouts due:

- [ ] Contributor: $500 @Tony-MK 
- [ ] Contributor+: $500 @DylanDylann

Upwork job is [here](https://www.upwork.com/jobs/~01bd7a8a1a0e4bc08b).

**Comment by DylanDylann** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 [@DylanDylann] The PR that introduced the bug has been identified. Link to the PR: **NA** 
 [@DylanDylann] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **NA**
 [@DylanDylann] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **NA**
 [@DylanDylann] Determine if we should create a regression test for this bug. **Yes**
 [@DylanDylann] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.


Regression Test Proposal
1. Go to a chat that has a video that was not uploaded by the device.
2. Open the video's context menu by holding down or left-click.
3. Verify that "Download" is a menu option.
4. Verify "Copy to clipboard" and "Edit comment" are not menu options.
5. Upload a video.
6. Open the video's context menu by holding down or left-click.
7. Verify that "Download" is a menu option.
8. Verify "Copy to clipboard" and "Edit comment" are not menu options.


Do we agree 👍 or 👎
